src/modules/category/adapter/controller/category.controller.ts

```diff
@@ -0,0 +1,40 @@
+import { Body, Controller, Delete, Get, Param, Post, UseGuards } from '@nestjs/common';
+import { CategoryService } from '../../usecase/service/category.service';
+import { SupabaseGuard } from 'src/modules/auth/guard/auth.guard';
+import { CategoryInput } from '../dto/category.input';
+import { CategoryOutput } from '../dto/category.output';
+
+@Controller('/categories')
+export class CategoryController {
+    constructor(private readonly categoryService: CategoryService) {}
+
+    @Post()
+    @UseGuards(SupabaseGuard)
+    async createCategory(@Body() category: CategoryInput): Promise<void> {
+        await this.categoryService.createCategory(category);
+    }
+
+    @Post('/:id')
+    @UseGuards(SupabaseGuard)
+    async updateCategory(@Param('id') id: string, @Body() category: CategoryInput): Promise<void> {
+        await this.categoryService.updateCategory(id, category);
+    }
+
+    @Get()
+    @UseGuards(SupabaseGuard)
+    async fetchCategories(): Promise<CategoryOutput[]> {
+        return await this.categoryService.getCategories();
```

Comment:
@PCalape Can you temporarily make the other APIs not callable by anyone, for the initial version of the app we will create an initial list of categories that users can use. In the second version that's when we will allow users to create their own categories.

For the initial version, let's keep all these APIs but make sure that users can only call the `GET /categories` and `GET /categories/:id`  APIs.

Reply:
Hello @arnellebalane sure will apply this one. Thank you!

Reply:
Hello @arnellebalane applied the suggestion and also refactored the environment variables for easier usage. Thanks!